File: 051/8.js
// Superclass Vehicle
class Vehicle {
    constructor(model, brand, price) {
      this._model = model; // Encapsulation: Protected attribute
      this._brand = brand; // Encapsulation: Protected attribute
      this._price = price; // Encapsulation: Protected attribute
    }
  
    // Getter and Setter for model
    get model() {
      return this._model;
    }
  
    set model(model) {
      this._model = model;
    }
  
    // Getter and Setter for brand
    get brand() {
      return this._brand;
    }
  
    set brand(brand) {
      this._brand = brand;
    }
  
    // Getter and Setter for price
    get price() {
      return this._price;
    }
  
    set price(price) {
      this._price = price;
    }
  
    // Method to display vehicle details
    displayDetails() {
      console.log(`รุ่น: ${this._model}`);
      console.log(`ยี่ห้อ: ${this._brand}`);
      console.log(`ราคา: ${this._price}`);
    }
  }
  
  // Subclass Car
  class Car extends Vehicle {
    constructor(model, brand, price, carType) {
      super(model, brand, price);
      this._carType = carType; // Encapsulation: Protected attribute
    }
  
    // Override displayDetails() for Car
    displayDetails() {
      super.displayDetails();
      console.log(`ประเภทของรถ: ${this._carType}`);
    }
  
    // Getter and Setter for carType
    get carType() {
      return this._carType;
    }
  
    set carType(carType) {
      this._carType = carType;
    }
  }
  
  // Subclass Bike
  class Bike extends Vehicle {
    constructor(model, brand, price, bikeType) {
      super(model, brand, price);
      this._bikeType = bikeType; // Encapsulation: Protected attribute
    }
  
    // Override displayDetails() for Bike
    displayDetails() {
      super.displayDetails();
      console.log(`ประเภทของจักรยาน: ${this._bikeType}`);
    }
  
    // Getter and Setter for bikeType
    get bikeType() {
      return this._bikeType;
    }
  
    set bikeType(bikeType) {
      this._bikeType = bikeType;
    }
  }
  
  // การใช้งาน
  // สร้าง Object ของ Vehicle
  const vehicle = new Vehicle("V1000", "Honda", 50000);
  vehicle.displayDetails();
  
  console.log("\n");
  
  // สร้าง Object ของ Car
  const car = new Car("Model S", "Tesla", 7000000, "Sedan");
  car.displayDetails();
  
  console.log("\n");
  
  // สร้าง Object ของ Bike
  const bike = new Bike("CBR600", "Yamaha", 150000, "Sport");
  bike.displayDetails();
  

File: 051/8.py

class Vehicle:
    def __init__(self, model, brand, price):
        self.__model = model  
        self.__brand = brand  
        self.__price = price  

 
    def get_model(self):
        return self.__model

    def set_model(self, model):
        self.__model = model


    def get_brand(self):
        return self.__brand

    def set_brand(self, brand):
        self.__brand = brand

    def get_price(self):
        return self.__price

    def set_price(self, price):
        self.__price = price

 
    def display_details(self):
        print(f"รุ่น: {self.__model}")
        print(f"ยี่ห้อ: {self.__brand}")
        print(f"ราคา: {self.__price}")


class Car(Vehicle):
    def __init__(self, model, brand, price, car_type):
        super().__init__(model, brand, price)
        self.__car_type = car_type

    def display_details(self):
        super().display_details()
        print(f"ประเภทของรถ: {self.__car_type}")

    def get_car_type(self):
        return self.__car_type

    def set_car_type(self, car_type):
        self.__car_type = car_type


class Bike(Vehicle):
    def __init__(self, model, brand, price, bike_type):
        super().__init__(model, brand, price)
        self.__bike_type = bike_type


    def display_details(self):
        super().display_details()
        print(f"ประเภทของจักรยาน: {self.__bike_type}")

    def get_bike_type(self):
        return self.__bike_type

    def set_bike_type(self, bike_type):
        self.__bike_type = bike_type


vehicle = Vehicle("V1000", "Honda", 50000)
vehicle.display_details()

print("\n")


car = Car("Model S", "Tesla", 7000000, "Sedan")
car.display_details()

print("\n")

# สร้าง Object ของ Bike
bike = Bike("CBR600", "Yamaha", 150000, "Sport")
bike.display_details()
